### ISSUE TITLE
composite priors do not work in stores

### ISSUE THREAD

**Issue by elidub** [undark-lab/swyft#96]
The composite priors defined as in the [tutorial notebook ](https://github.com/undark-lab/swyft/blob/master/notebooks/Examples%20-%207.%20Composite%20Priors.ipynb) don't work in stores. Changing priors in tutorial notebooks with 

``` 
from swyft.utils import tensor_to_array, array_to_tensor
from toolz import compose

uniform = torch.distributions.Uniform(array_to_tensor(low), array_to_tensor(high))
prior = swyft.Prior(
        cdf = compose(tensor_to_array, lambda x: uniform.cdf(x), array_to_tensor),
        icdf = compose(tensor_to_array, lambda x: uniform.icdf(x), array_to_tensor),
        log_prob = compose(tensor_to_array, lambda x: uniform.log_prob(x), array_to_tensor),
        n_parameters = n_parameters,
    )
```

Gives the following error: `PicklingError: Can't pickle <function <lambda> at 0x1530b34e7e50>: attribute lookup <lambda> on __main__ failed`

**Comment by bkmi** [comment]
Thank you for reporting the issue. I'll fix it before the 0.3.0 release.

The reason for this is due to how pickle works. (It cannot pickle lambda functions)

**Comment by elidub** [comment]
Works great!
